Conduit facades

Rover656 opened the pull request:
# Description

- Conduit Facades (normal, transparent, hardened and transparent hardened).
- Holding any wrench in main hand or offhand hides the facades and allows interaction under the hood.

Deferred for future:
- Enable/disable this behaviour with shift+scroll on Yeta wrench
- Allow showing/hiding individual conduit types

Closes #301 

# TODO

- [x] Blast resistance increase
- [ ] ~~Ability to place conduit bundle with no conduits (facade only)~~ future work
- [x] Fix AE2 conduit facade interop
- [x] Ensure block break interactions are correct
- [x] Render wireframe (possibly with a BER) when the facades are hidden
- [x] Fix collision work when hiding facades (should still collide like a full block)
- [ ] ~~Config option for enable/disable conduit facade hiding functionality entirely.~~ cannot be a user preference with how it is designed atm.

<!-- For drafts, fill this in as you go; if you are leaving draft, make sure these are all complete. -->
# Checklist

- [x] My code follows the style guidelines of this project (.editorconfig, most IDEs will use this for you).
- [x] I have made corresponding changes to the documentation.
- [x] My changes are ready for review from a contributor.

<!-- Thanks to: https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ for the building blocks of this template -->

Rover656 committed:
feat: Initial enablement of facades. Allows a single block type and supports the "transparent" variant (in code). Facades hide when holding a wrench.

Next need to add facade items instead of using the painted slab, handle increased blast resistance and allow facades to be placed without any conduit in the space (for laying down future paths). Also need to handle break interactions.

Rover656 committed:
feat: Added facade items and facade breaking behaviour.

HenryLoenwind commented on enderio-conduits/src/main/java/com/enderio/conduits/client/model/conduit/facades/FacadeChunkReloadHelper.java:
```diff
@@ -0,0 +1,36 @@
+    public static void onEquipmentChanged(LivingEquipmentChangeEvent event) {
+        if (event.getSlot() != EquipmentSlot.MAINHAND &&
+            event.getSlot() != EquipmentSlot.OFFHAND) {
+            return;
```
`|| (!event.getEntity() instanceof LocalPlayer)` is a shorter check than what happens next.

Rover656 replied:
Good call, thanks!

Rover656 committed:
chore: Some cleanup and fixed collisions when hiding facades with a wrench.

Rover656 committed:
fix: Allow placing conduits into an existing bundle when right clicking the block below

turnerlewis commented:
You're doing gods work guys!

Rover656 committed:
chore: change how facades are stored and tidy up after the rebase.

Conduit items now also hide facades

Rover656 committed:
chore: Working out how we might texture the facades.

Currently invisible before painting now.

ferriarnus committed:
reworked facade handeling

- transparent rendering
- custom item model
- CTM and tint support
- Item models

Rover656 committed:
fix: Misunderstood old transparent facade type

Rover656 committed:
fix: Facade visibility check doesn't work when switching yeta from offhand to main.

Rover656 committed:
feat: Simple tooltip explaining the facade types.

Rover656 committed:
chore: spotlessApply

Rover656 committed:
feat: AE2 Facade Support

Rover656 committed:
chore: Add tooltip to our facade items only to avoid polluting other's tooltips.

Rover656 committed:
chore: spotlessApply

Rover656 committed:
fix: Remove facades from the client map when unloaded

Rover656 committed:
chore: Expose facade provider logic via the API.

Rover656 committed:
chore: spotlessApply

Rover656 committed:
feat: Change facade overlay to similar texture to 1.12.2
